File: apple/Sources/ReactNativeWatchOS/Modifiers/OnAppear.swift
import SwiftUI

/// SwiftUI `.onAppear(perform:)`. Fires the JS `handler` when the view appears.
enum RNWOnAppearModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("onAppear") { view, params, ctx in
            AnyView(view.onAppear { ctx.fire(params.int("handler")) })
        }
    }
}

File: src/modifiers/onAppear.ts
import { createModifier } from './createModifier';

/// SwiftUI `.onAppear(perform:)`. Fires `handler` when the view appears.
///
/// ```tsx
/// <Text modifiers={[onAppear(() => track('seen'))]} />
/// ```
export function onAppear(handler: () => void) {
  return createModifier('onAppear', { handler });
}
